### ISSUE TITLE
[core][rlgl] TakeScreenshot() not working as expected

### ISSUE THREAD

**Issue by raysan5** [raysan5/raylib#36]
When taking a screenshot, if some transparent element has been drawn, returned PNG image shows the transparent pixels.

Related functions [core.c][rlgl.c]
 
    static void TakeScreenshot(void);
    unsigned char *rlglReadScreenPixels(int width, int height);

Framebuffer color data is retrieved with `glReadPixels()` as GL_RGBA. Despite we don't see the transparent information in the running window, transparent info it's there... But, probably, a non-transparent image is expected by the user...

**Comment by raysan5** [comment]
After further investigation of this issue and long reads about pre-multiplied alpha and the way OpenGL manages blending, I got to the conclusion that the returned image screenshot is correct, despite the fact that probably user won't expect a transparent screenshot because it's not what they see in the running window.

Posible tweak: Process the returned backbuffer color data to get the expected (non-alpha) result.

After some tries trying to post-multiply alpha, I couldn't get it to work correctly...

**Comment by raysan5** [comment]
After further investigation and testing, I realized that colors are blended correctly and alpha is just useless (it has already been applied to color!).

Useful reference: http://www.andersriggelsen.dk/glblendfunc.php

Let's understand the used OpenGL blending function (additive equation):

     glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);

It translates to the following equation:

    finalColor = sourceColor*sourceAlpha + destinationColor*(1 - sourceAlpha)

Translated to specific color components:

    finalColor.r = sourceColor.r*sourceColor.a + destinationColor.r*(1 - sourceColor.a)
    finalColor.g = sourceColor.g*sourceColor.a + destinationColor.g*(1 - sourceColor.a)
    finalColor.b = sourceColor.b*sourceColor.a + destinationColor.b*(1 - sourceColor.a)
    finalColor.a = sourceColor.a*sourceColor.a + destinationColor.a*(1 - sourceColor.a)   // Issue!

As we saw, sourceColor.a is already applied to calculate finalColor.rgb. As we our color framebuffer is RGBA include finalColor.a, the same function is also applied for the alpha component. That's useless for us and just adds and undesired transparency to the result captured RGBA image.

Solution: Just set `finalColor.a = 255` to the result image; it can be done at the same time we flip the image vertically.

Solution was applied in changelist https://github.com/raysan5/raylib/commit/5208d57f1e0180ea095cecec64d3db7703ec63c9
